include/core/CoreTypes.h

```diff
@@ -17,17 +17,12 @@ using TTime = time_t;
 
 //! The standard line ending for the platform - DON'T make this std::string as
 //! that would cause many strings to be constructed (since the variable is
-//! static const at the namespace level, so is internal to each file this
-//! header is included in)
+//! const at the namespace level, so is internal to each file this header is
+//! included in)
 #ifdef Windows
-static const char* LINE_ENDING = "\r\n";
+const char* const LINE_ENDING = "\r\n";
 #else
-#ifdef __GNUC__
-// Tell g++ that it's reasonable that this variable isn't used
-__attribute__((unused)) static const char* LINE_ENDING = "\n";
```

Comment:
Not needed?

Reply:
No, now the variable really does have internal linkage it's not needed.

In fact, the warning from g++ was the canary in the mine warning that something was not right.  Suppressing the warning was the wrong way to avoid it.